src/app/(auth)/subscriber/page.tsx

```diff
@@ -1,54 +1,75 @@
 'use client';
 
 import Button from '@/components/Button';
-import { getSubscribers } from '@/apis/subscriber';
-import Input from '@/components/Input';
+import { getSubscribers, deleteSubscriber } from '@/apis/subscriber';
 import Table from '@/components/Table';
 import { useEffect, useState } from 'react';
-import { Subscribers, TableData } from '@/utils/types';
+import { SubscriberResource, TableData } from '@/utils/types';
 import { UserGroupIcon } from '@heroicons/react/24/outline';
 import { PencilSquareIcon, TrashIcon } from '@heroicons/react/24/solid';
+import Link from 'next/link';
+import { toast } from 'react-toastify';
 
 const Page = () => {
-  const [subscribers, setSubscribers] = useState<Subscribers | undefined>(
-    undefined
-  );
+  const [subscribers, setSubscribers] = useState<
+    SubscriberResource | undefined
+  >(undefined);
 
-  useEffect(() => {
-    const fetchSubscribers = async () => {
-      setSubscribers(await getSubscribers());
-    };
-    fetchSubscribers();
-  }, []);
+  const fetchSubscribers = async () => {
+    setSubscribers(await getSubscribers());
+  };
 
   const [tableData, setTableData] = useState<TableData>({
     headers: ['Details', 'List'],
     rows: [],
   });
 
   useEffect(() => {
+    fetchSubscribers();
+    const showDeleteConfirmationModal = async (id: number) => {
+      const result = confirm(
+        'Are you sure you want to delete this subscriber?'
+      );
+      if (result) {
+        try {
+          await deleteSubscriber(id);
+          toast.success('Subscriber deleted successfully');
+          fetchSubscribers();
+        } catch (error: any) {
+          toast.error(error.response.data.message);
+        }
+      }
+    };
     if (subscribers) {
-      const rows = subscribers.data.map((subscriber: Subscribers) => {
+      const rows = subscribers.data.map((subscriber) => {
```

Comment:
why removed type from here?

Reply:
We added this while creating the subscriber page and didn't cleanup the code at that time. Now we do not need it, so I have removed the unused code in it.

Reply:
We should use and define types wherever possible. This is a typescript app.